auth-service/src/main/java/com/kioke/auth/model/User.java

```diff
@@ -4,19 +4,38 @@
 import jakarta.persistence.Entity;
 import jakarta.persistence.Id;
 import jakarta.persistence.Table;
+import java.util.Collection;
 import lombok.AllArgsConstructor;
 import lombok.Builder;
 import lombok.Data;
 import lombok.NoArgsConstructor;
+import org.hibernate.annotations.UuidGenerator;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
 
 @Entity
 @Table(name = "USER_TABLE")
 @Data
 @AllArgsConstructor
 @NoArgsConstructor
 @Builder
-public class User {
+public class User implements UserDetails {
   @Id
-  @Column(name = "uid")
+  @UuidGenerator
+  @Column(name = "uid", unique = true, nullable = false)
   private String uid;
+
+  private String email;
```

Comment:
The user model will have more read accesses than write accesses. Let's make the email field an indexed column to improve performance.